src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -479,15 +479,15 @@ public override int ExecuteNonQuery()
             {
                 throw new InvalidOperationException(Resources.CallRequiresOpenConnection(nameof(ExecuteNonQuery)));
             }
-            if (_commandText == null)
+            if (string.IsNullOrWhiteSpace(_commandText))
             {
                 throw new InvalidOperationException(Resources.CallRequiresSetCommandText(nameof(ExecuteNonQuery)));
             }
 
-            var reader = ExecuteReader();
-            reader.Dispose();
-
-            return reader.RecordsAffected;
+            using (var reader = ExecuteReader())
+            {
+                return reader.RecordsAffected;
+            }
```

Comment:
Hmm, this might result in a different value. `RecordsAffected` may not include all the results until all the result sets are consumed.

Reply:
(It certainly will after we fix #422)

Reply:
Well, this just make your codes in union like what you see line at 509~515, which uses `using....` statement to wrap those implementing IDisposible.